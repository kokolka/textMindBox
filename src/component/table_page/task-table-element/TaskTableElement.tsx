import React from 'react';

const TaskTableElement: React.FC<{s: string}> = ({s}) => {
    return(
        <div>

        </div>
    )
}

export default TaskTableElement;