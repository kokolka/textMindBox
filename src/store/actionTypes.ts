export const TASK_TABLE_ADD = 'taskTable/addTask';
export const TASK_TABLE_PERFORM = 'taskTable/addPerform';