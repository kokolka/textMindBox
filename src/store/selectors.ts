import {StateType} from './store';


export const getTaskSelector = (state: StateType) => state.taskReducer.task;

export const getAppStateSelector = (state: StateType) => state.stateAppReducer.formForTask;