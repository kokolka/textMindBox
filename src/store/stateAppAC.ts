import { CLOSE_FORM_FOR_TASK, OPEN_FORM_FOR_TASK } from "./actionTypes"


export type openFormForTaskType = {
    type: typeof OPEN_FORM_FOR_TASK
}
export const openFormForTask = (): openFormForTaskType => ({type: OPEN_FORM_FOR_TASK});

export type closeFormForTaskType = {
    type: typeof CLOSE_FORM_FOR_TASK
}
export const closeFormForTask = (): closeFormForTaskType => ({type: CLOSE_FORM_FOR_TASK});