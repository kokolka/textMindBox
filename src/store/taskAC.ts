import { TASK_TABLE_ADD, TASK_TABLE_PERFORM } from './actionTypes';

export type addTaskType = {
    type: typeof TASK_TABLE_ADD
    payload: string
}
export const addTask = (text: string): addTaskType => ({type: TASK_TABLE_ADD, payload: text})

export type performTaskType = {
    type: typeof TASK_TABLE_PERFORM
    payload: number
}
export const performTask = (id: number): performTaskType => ({type: TASK_TABLE_PERFORM, payload: id})