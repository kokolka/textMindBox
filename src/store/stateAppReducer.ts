import { ACTIVE_TASK, ALL_TASK, COMPLETED_TASK} from "./actionTypes";
import { getActiveTaskType, getAllTackType, getCompletedTaskType } from "./stateAppAC"

type stateAppACType = getAllTackType | getActiveTaskType | getCompletedTaskType;

export type formForTaskType = 'all' | 'active' | 'completed'; //экспорт типов для тестирования 

const initialState = {
    formForTask: 'all' as formForTaskType
}

type StateType = typeof initialState;

const stateAppReducer = (state = initialState, action: stateAppACType): StateType => {
    switch(action.type){
        case ALL_TASK: {
            return {
                ...state,
                formForTask: 'all'
            }
        }
        case ACTIVE_TASK: {
            return {
                ...state,
                formForTask: 'active'
            }
        }
        case COMPLETED_TASK: {
            return {
                ...state,
                formForTask: 'completed'
            }
        }
        default: return state
    }
}

export default stateAppReducer;