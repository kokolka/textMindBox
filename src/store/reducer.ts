import { addTaskType } from './taskAC';

type taskType = {
    text: string,
    id: number,
    perform: boolean
}

const initialState = {
    task: [

    ] as Array<taskType>
}

export const addTaskReducer = (state = initialState, action: addTaskType) => {

    let lastId = 0;
    if(state.task.length !== 0){
        lastId = state.task[state.task.length - 1].id + 1;
    }

    const newTask = {
        id: lastId,
        text: action.payload, 
        perform: false
    } as taskType

    return {
        ...state,
        task: [...state.task, newTask]
    }
}