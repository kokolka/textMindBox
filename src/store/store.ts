import taskReducer from './taskReducer';
import stateAppReducer from './stateAppReducer';
import { legacy_createStore as createStore, combineReducers } from 'redux';
import { addTaskType, performTaskType } from './taskAC';
import { getActiveTaskType, getAllTackType, getCompletedTaskType } from './stateAppAC';

type storeACType =
  | addTaskType
  | performTaskType
  | getAllTackType
  | getActiveTaskType
  | getCompletedTaskType;

const reducers = combineReducers({
  tasks: taskReducer,
  selectorActiveTasks: stateAppReducer,
});

type reducersType = typeof reducers;
export type StateType = ReturnType<reducersType>; //тип глобального state

export const store = createStore<any, storeACType, unknown, unknown>(reducers);
