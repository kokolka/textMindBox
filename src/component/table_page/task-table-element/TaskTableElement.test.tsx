import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { StateType } from '../../../store/store';
import { TaskTableElement } from './TaskTableElement';

const mockStore = configureStore<StateType>([]);

describe('Тестирование отрисовки блока с кнопками', () => {
  let store: ReturnType<typeof mockStore>;

  beforeEach(() => {
    store = mockStore({
      tasks: {
        taskIdCounter: 1,
        task: [{ text: 'Создать первую задачу', id: 0, perform: false }],
      },
      selectorActiveTasks: {
        formForTask: 'all',
      },
    });
  });

  test('renders with initial state', () => {
    const component = renderer.create(
      <Provider store={store}>
        <TaskTableElement />
      </Provider>,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
