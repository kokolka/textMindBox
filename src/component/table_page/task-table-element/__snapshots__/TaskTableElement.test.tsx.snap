// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование отрисовки блока с кнопками renders with initial state 1`] = `<div />`;
